Solar-Wrapper/src/TrackablesSolARImpl.cpp

```diff
@@ -0,0 +1,287 @@
+            {
+                response.headers().add<Pistache::Http::Header::ContentType>(MIME(Text, Plain));
+                response.send(Pistache::Http::Code::Not_Found, "Trackable not found\n");
+                break;
+            }
+
+            default :
+            {
+                response.headers().add<Pistache::Http::Header::ContentType>(MIME(Text, Plain));
+                response.send(Pistache::Http::Code::Internal_Server_Error, "Something went wrong\n");
+            }
+        }
+    }
+
+    void TrackablesSolARImpl::get_trackable_by_id(const std::string &trackableUUID, Pistache::Http::ResponseWriter &response)
+    {
+        //initialize the json object that we will send back to the client
+        auto jsonObjects = nlohmann::json::array();
+
+        //look for the trackable with given id
+        xpcf::uuids::uuid id = xpcf::toUUID(trackableUUID);
+        SRef<StorageTrackable> storageTrackable;
+
+        switch(m_worldStorage->getTrackable(id, storageTrackable))
+        {
+            case FrameworkReturnCode::_SUCCESS :
+            {
+                //StorageTrackable found, we convert it into a Trackable
+                Trackable trackable = fromStorage(*storageTrackable);
+
+                //add the Trackable to our JSON object
+                to_json(jsonObjects, trackable);
+
+                //send the Trackable to the client
+                response.headers().add<Pistache::Http::Header::ContentType>(MIME(Application, Json));
+                response.send(Pistache::Http::Code::Ok, jsonObjects.dump());
+
+                break;
+            }
+
+            case FrameworkReturnCode::_NOT_FOUND :
+            {
+                response.headers().add<Pistache::Http::Header::ContentType>(MIME(Text, Plain));
+                response.send(Pistache::Http::Code::Not_Found, "Trackable not found\n");
+                break;
+            }
+
+            default :
+            {
+                response.headers().add<Pistache::Http::Header::ContentType>(MIME(Text, Plain));
+                response.send(Pistache::Http::Code::Internal_Server_Error, "Something went wrong\n");
+            }
+        }
+
+
+    }
+
+    void TrackablesSolARImpl::get_trackables(Pistache::Http::ResponseWriter &response)
+    {
+
+        //initialize the json object that we will send back to the client
+        auto jsonObjects = nlohmann::json::array();
+
+        //declaration of all the objects that will be changed at each iteration of the loop
+        nlohmann::json toAdd;
+        Trackable track;
+
+        std::vector<SRef<datastructure::StorageTrackable>> vector;
+        if(m_worldStorage->getTrackables(vector) != FrameworkReturnCode::_SUCCESS)
+        {
+            //Exception raised in getTrackables
+            response.headers().add<Pistache::Http::Header::ContentType>(MIME(Text, Plain));
+            response.send(Pistache::Http::Code::Internal_Server_Error, "Something went wrong when fetching the world storage");
```

Comment:
Is the message correct ? The trackables are fetched, not the world storage. "When fetching the trackables from the world storage", is that what you meant ?